legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalGeneratorAbstract.java

```diff
@@ -225,6 +225,11 @@ GeneratorResult generateOperations(Datasets datasets, Resources resources)
 
     GeneratorResult generateOperations(Datasets datasets, Resources resources, Planner planner, IngestMode ingestMode)
     {
+        if (!relationalSink().isIngestModeSupported(ingestMode))
```

Comment:
Should we do it even earlier -- may be like a validation using @Check  ?

Reply:
Sure great idea, changed!